services/federation/ap.go

```diff
@@ -0,0 +1,24 @@
+package federation
+
+import "github.com/yitsushi/go-misskey/core"
+
+// ShowRequest is the request used by the show command.
+// Contains a single URI.
+type ShowRequest struct {
+	URI string `json:"uri"`
+}
+
+// ShowResponse is an empty response for the Show command.
```

Comment:
I'm pretty sure it's not empty, the documentation says "without any results", but they have. After #29 is merged, it will be much easier to check what's the response (and I'll document how).

Reply:
```
# Response if you ask a about a Misskey AP entity
curl -X POST -d '{"uri": "https://slippy.xyz/notes/8dmmpkv4lo"}' https://slippy.xyz/api/ap/show | jq

# Response if you ask about a Mastodon AP entity
curl -X POST -d '{"uri": "https://mastodon.art/@Krita/105027075106876908"}' https://slippy.xyz/api/ap/show | jq
```

It's the same, but more sample, easier to understand.

Reply:
Oh sh!t, maybe just ignore this endpoint for now...

We have to split this function up, because it has different responses for different kind of resources, so here we need a custom parser to parse only the Type and based on the Type, parse the rest.

```
# Asking about a user
curl -X POST -d '{"uri": "https://mastodon.art/@Krita"}' https://slippy.xyz/api/ap/show | jq
```

Reply:
Affirmative. :)